src/backend/ImageBackend.cc

```diff
@@ -37,9 +37,17 @@ int32_t ImageBackend::approxBytesPerPixel() {
 
 cairo_surface_t* ImageBackend::createSurface()
 {
-	assert(!this->surface);
-	this->surface = cairo_image_surface_create(this->format, width, height);
-	assert(this->surface);
+	assert(!surface);
+	surface = cairo_image_surface_create(format, width, height);
+
+  if (!surface)
+    return NULL;
+
+  if (width && height && !cairo_image_surface_get_data(surface)) {
```

Comment:
Is this `cairo_image_surface_get_data` check necessary? The docs say it returns:

> a pointer to the image data of this surface or NULL if surface is not an image surface, or if cairo_surface_finish() has been called.

which are not conditions that I think we need to worry about here.

It sounds like checking `surface` as you've done above is sufficient:

> `cairo_image_surface_create` always returns a valid pointer, but it will return a pointer to a "nil" surface if an error such as out of memory occurs